src/shared/gcpagent/utils.go

```diff
@@ -0,0 +1,102 @@
+package gcpagent
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/iam/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const truncatedHashLength = 6
+
+const maxK8SNameLength = 250
+const maxK8STruncatedLength = maxK8SNameLength - truncatedHashLength - 1 // add another char for the hyphen
+
+const maxDisplayNameLength = 100
+const maxDisplayNameTruncatedLength = maxDisplayNameLength - truncatedHashLength - 1 // add another char for the hyphen
+
+const maxGCPNameLength = 30
+const maxGCPTruncatedLength = maxGCPNameLength - truncatedHashLength - 1 // add another char for the hyphen
+
+func (a *Agent) limitResourceName(name string, maxLength int) string {
+	var truncatedName string
+	if len(name) >= maxLength {
+		truncatedName = name[:maxLength]
+	} else {
+		truncatedName = name
+	}
+
+	hash := fmt.Sprintf("%x", sha256.Sum256([]byte(name)))
+	hash = hash[:truncatedHashLength]
+
+	return fmt.Sprintf("%s-%s", truncatedName, hash)
+}
+
+func (a *Agent) generateKSAPolicyName(namespace string, intentsServiceName string) string {
+	name := fmt.Sprintf("otr-%s-%s-intent-policy", namespace, intentsServiceName)
+	return a.limitResourceName(name, maxK8STruncatedLength)
+}
+
+func (a *Agent) generateGSAToKSAPolicyName(ksaName string) string {
+	name := fmt.Sprintf("otr-%s-gcp-identity", ksaName)
+	return a.limitResourceName(name, maxK8STruncatedLength)
+}
+
+func (a *Agent) generateGSADisplayName(namespace string, accountName string) string {
+	name := fmt.Sprintf("otr-%s-%s-%s", a.clusterName, namespace, accountName)
+	return a.limitResourceName(name, maxDisplayNameTruncatedLength)
+}
+
+func (a *Agent) generateGSAName(namespace string, accountName string) string {
+	fullName := a.generateGSADisplayName(namespace, accountName)
+	return a.limitResourceName(fullName, maxGCPTruncatedLength)
+}
+
+func (a *Agent) GetGSAFullName(namespace string, accountName string) string {
```

Comment:
Should this name (and any other name that exists in the GCP project scope) take into account the cluster name? What happens if the same (service account name, namespace) exist in two clusters in the same GCP account?